packages/sdk/src/utils/transaction.ts

```diff
@@ -17,9 +17,11 @@ export async function sendTransactionSafely(
   const transaction = await signer.sendTransaction({
     chain: viemChain,
     data: data.data,
-    account: data.from,
+    account: signer.account ?? data.from, // use signer.account if it's defined
     to: data.to,
-    value: data.value,
+    value: hexToBigInt(data.value),
+    ...(data.maxFeePerGas && {maxFeePerGas: hexToBigInt(data.maxFeePerGas)}),
+    ...(data.maxPriorityFeePerGas && {maxPriorityFeePerGas: hexToBigInt(data.maxPriorityFeePerGas)}),
```

Comment:
viem requires [`bigInt` for value & gas fees](https://viem.sh/docs/actions/wallet/sendTransaction.html#value-optional), but reservoir was previously passing hex values.
also added `maxFeePerGas` & `maxPriorityFeePerGas`, since those are optional params in `buyToken`